Document changes to annotation examples (#39283)
openshift/openshift-docs: modules/sbo-categories-of-exposable-binding-data.adoc

----

== Exposing a constant value as the binding item

> The following examples show how to expose constant value from the `PostgresCluster` custom resource (CR):

Reviewer: ```suggestion
The following examples show how to expose a constant value from the `PostgresCluster` custom resource (CR):
```

> .Example: Exposing constant value

Reviewer: ```suggestion
.Example: Exposing a constant value
```

[source,yaml]
----
apiVersion: postgres-operator.crunchydata.com/v1beta1
kind: PostgresCluster
metadata:
  name: hippo
  namespace: my-postgresql
  annotations:
    "service.binding/type": "postgresql" <1>
----

> <1> Binding `type` that is to be exposed with the `postgresql` value.

Reviewer: ```suggestion
<1> Binding `type` to be exposed with the `postgresql` value.
```
Consider omitting `that is` to tighten up phrasing.

----

== Exposing a constant value as the binding item
The following examples show how to expose constant value as annotations:

> .Example: Exposing constant value as annotations

Reviewer 2: I would drop `as annotations`, and leave just `Exposing constant value`

Author: Addressed.

Reviewer 3: @Srivaralakshmi this drop of `as annotations` would be applied to all example titles.

[source,yaml]
----
apiVersion: postgres-operator.crunchydata.com/v1beta1
kind: PostgresCluster
metadata:
  name: hippo
  namespace: my-postgresql
  annotations:

>     "service.binding/type": "postgresql" <1>

Reviewer 4: I don't think we can directly specify a string value. See the example: https://docs.openshift.com/container-platform/4.9/applications/connecting_applications_to_services/getting-started-with-service-binding.html

Reviewer 2: Yes, we can - this was added as a part of https://github.com/redhat-developer/service-binding-operator/pull/1040

Author: I believe there is no change to be done here. @pedjak please confirm.

The following example shows how to expose a resource definition value as OLM descriptors:

> .Example: Exposing a resource definition value as OLM descriptors

Reviewer 3: ```suggestion
.Example: Exposing a resource definition value using OLM descriptors
```

Reviewer 3: That `as` -> `using` suggesting would to all example titles where `Exposing ... as ...` is.

Author: As discussed replaced all the instances with "through" to maintain consistency across the usage of words.

== Exposing entries of a collection with the key and value from each entry

> Following is the example for exposing the entries of a collection with the key and value from each entry through annotations:

Reviewer: ```suggestion
The following example shows how to expose the entries of a collection with the key and value from each entry through annotations:
```
Rephrase to be consistent with phrasing used in prior examples

/bindings/<binding-name>/uri_404 => black-hole.example.com
----

> Following is the example for exposing the entries of a collection with the key and value from each entry through OLM descriptors:

Reviewer: ```suggestion
The following example shows how to expost the entries of a collection with the key and value from each entry through OLM descriptors:
```
Rephrase to be consistent with phrasing used in prior examples.

Reviewer 3: There is a typo in the suggestion `expost`, should be `expose`.

Author: Thanks a lot, @pmacik, and @bburt-rh!

== Exposing items of a collection with one key per item

> Following is the example for exposing the items of a collection with one key per item through annotations:

Reviewer: ```suggestion
The following example shows how to expose the items of a collection with one key per item through annotations:
```
Rephrase to be consistent with phrasing used in prior examples

/bindings/<binding-name>/tags_2 => power
----

> Following is the example for exposing the items of a collection with one key per item through OLM descriptors:

Reviewer: ```suggestion
The following example shows how to expose the items of a collection with one key per item through OLM descriptors:
```
Rephrase to be consistent with phrasing used in prior examples

== Exposing values of collection entries with one key per entry value

> Following is the example for exposing the values of collection entries with one key per entry value through annotations:

Reviewer: ```suggestion
The following example shows how to expose the values of collection entries with one key per entry value through annotations:
```
Rephrase to be consistent with phrasing used in prior examples.

/bindings/<binding-name>/url_2 => black-hole.example.com
----

> Following is the example for exposing the values of collection entries with one key per entry value through OLM descriptors:

Reviewer: ```suggestion
The following example shows how to expose the values of collection entries with one key per entry value through OLM descriptors:
```
Rephrase to be consistent with phrasing used in prior examples
